lib/validation.js

```diff
@@ -9,8 +9,10 @@ const joi = require('joi');
 /**
  * Validate user input against rules
  *
- * @param {function} rule - Rule created with the joi validation framework
- * @returns {Error|boolean} return true if input is valid, return the error otherwise
+ * @param {function} rule - Validation rule created with the joi validation framework
+ * @returns {function} calls an anonymous function that returns either
+ * - an error message {Error} in case of error,
```

Comment:
Despite it being semantically an error, the error message is of type string and not Error.